grafana/grafana
# InfluxDB: InfluxQL: query editor: skip fields in metadata queries

in the influxdb influxql query editor, the WHERE section is primarily meant to be used with influxdb_tags. but it is possible to use influxdb_fields there too. it works fine for running the query, but it can cause problems for metadata queries:

if you have an influxdb_field in the WHERE part, and then try to add another segment to WHERE by clicking on the [+], we ask influxdb for possible tag_key values in the dropdown, and, because we have an influxdb_field in it, influxdb returns the "expression too complex for metaquery" error.

to fix this, when doing metadata queries, we only send real influxdb_tags to the database, we skip influxdb_fields. 

to differentiate between influxdb_tags and influxdb_fields, when the query editor starts, we ask influxdb for all tag_keys. with this info, we can then filter the tags+fields array to only have tags when doing a metadata query.

how to test:
- `make devenv sources=influxdb`
- wait 20 seconds
- go to explore, choose the `gdev-influxdb-influxql` data source
- in `select measurement` choose `cpu`
- below, in `field(value)` choose `usage_idle`
- click on the [+] next to the WHERE part and manually enter `usage_idle`
- set the filter to form `usage_idle = 42`
- now click on the `+` again. verify that you get a list of tag keys (`cpu` and `host`)
- choose `cpu`
- click on the `select tag value`
- verify that you get a list of options: `cpu-total`, `cpu0`, `cpu1`, `cpu2`, `cpu3`
- choose `cpu1`
- now open the `cpu` dropdown` in the FROM section
- verify that you get 1 option `cpu` (i mean, verify that you do not get zero options)
